### PR TITLE
Upgrade to tensorflow 2.0 - drop Keras dependency

### LINKED ISSUE #495: Upgrade to tensorflow 2
Tensorflow 2 is now out! Lets upgrade. And by doing that I think we can remove keras as a requirement?

### PR DESCRIPTION
**Pull request by milesgranger**
Will close #495

### CODE REVIEW

**Comment by epa095** [comment]
Sexy

File: gordo_components/model/models.py [hunk 1]
```diff
@@ -194,20 +155,30 @@ def get_params(self, **params):
...
-        with possible_tf_mgmt(self):
-            return build_fn(**self.sk_params)
+        return build_fn(**self.sk_params)
 
     def save_to_dir(self, directory: str):
         params = self.get_params()
         with open(path.join(directory, "params.json"), "w") as f:
             json.dump(params, f)
         if hasattr(self, "model") and self.model is not None:
-            with possible_tf_mgmt(self):
```
Comment (epa095) [inline gordo_components/model/models.py:205]: damn it feels good getting rid of this:-)

File: gordo_components/model/models.py [hunk 2]
```diff
@@ -194,20 +155,30 @@ def get_params(self, **params):
...
-                self.model.save(path.join(directory, "model.h5"))
-                if hasattr(self.model, "history") and self.model.history is not None:
-                    f_name = path.join(directory, "history.pkl")
-                    with open(f_name, "wb") as history_file:
-                        pickle.dump(self.model.history, history_file)
+            with open(path.join(directory, "model.json"), "w") as model_json:
+                json.dump(self.model.to_json(), model_json)
```
Comment (epa095) [inline gordo_components/model/models.py]: this is weird, why do you have to do json.dump on a json?

Comment (epa095) [reply gordo_components/model/models.py]: Also, do you want to do a quick benchmark of how long it takes to load a (big lstm) tensorflow model with the 5 alternatives:
- `model.save('path_to_saved_model', save_format='tf', include_optimizer=True)`
- `model.save('path_to_saved_model', save_format='h5', include_optimizer=True)`
- `model.save('path_to_saved_model', save_format='tf', include_optimizer=False)`
- `model.save('path_to_saved_model', save_format='h5', include_optimizer=False)`
- save model weight/arcitecture as json.

My guess is that the 3 last ones are pretty much the same.

**Comment by flikka** [comment]
So gøy!

<sub>Sent with <a href="http://githawk.com">GitHawk</a></sub>

File: gordo_components/model/models.py (hunk 2 above)
Comment (milesgranger) [reply gordo_components/model/models.py]: Results are in!
```
Save as JSON and h5 seperatly
-----
14.13s call     tests/gordo_components/model/test_model.py::test_save_load[KerasLSTMAutoEncoder-lstm_hourglass]
12.60s call     tests/gordo_components/model/test_model.py::test_save_load[KerasLSTMForecast-lstm_hourglass]
1.50s call     tests/gordo_components/model/test_model.py::test_save_load[KerasAutoEncoder-feedforward_hourglass]

Save as model.tf, include_optimizer=True, compile=True
-----
51.24s call     tests/gordo_components/model/test_model.py::test_save_load[KerasLSTMForecast-lstm_hourglass]
50.07s call     tests/gordo_components/model/test_model.py::test_save_load[KerasLSTMAutoEncoder-lstm_hourglass]
3.73s call     tests/gordo_components/model/test_model.py::test_save_load[KerasAutoEncoder-feedforward_hourglass]

Save as model.h5, include_optimizer=True, compile=True
-----
16.96s call     tests/gordo_components/model/test_model.py::test_save_load[KerasLSTMAutoEncoder-lstm_hourglass]
14.52s call     tests/gordo_components/model/test_model.py::test_save_load[KerasLSTMForecast-lstm_hourglass]
2.02s call     tests/gordo_components/model/test_model.py::test_save_load[KerasAutoEncoder-feedforward_hourglass]

Save as model.h5, include_optimizer=True, compile=False
-----
12.30s call     tests/gordo_components/model/test_model.py::test_save_load[KerasLSTMAutoEncoder-lstm_hourglass]
12.12s call     tests/gordo_components/model/test_model.py::test_save_load[KerasLSTMForecast-lstm_hourglass]
1.56s call     tests/gordo_components/model/test_model.py::test_save_load[KerasAutoEncoder-feedforward_hourglass]

Save as model.h5, include_optimizer=False, compile=False
-----
12.62s call     tests/gordo_components/model/test_model.py::test_save_load[KerasLSTMAutoEncoder-lstm_hourglass]
12.35s call     tests/gordo_components/model/test_model.py::test_save_load[KerasLSTMForecast-lstm_hourglass]
2.03s call     tests/gordo_components/model/test_model.py::test_save_load[KerasAutoEncoder-feedforward_hourglass]
```

Note: It was not possible to have `save_format='tf'` with `compile=False` _and/or_ `include_optimizer=False`

**Review by epa095 — COMMENTED** [review]
So much deleted code! Summary of our chat:
- Its OK to not be backwards compatible on the optimizer names. 
- Change to use keras save

**Review by epa095 — APPROVED** [review]
LGTM after squash

**Pull request merged**